backend/battle/tests/test_view.py

```diff
@@ -1,15 +1,104 @@
+        self.battle = baker.make('battle.Battle', creator=self.user)
+        self.battle_from_user = Battle.objects.filter(creator=self.user)
+
+    def test_login_user_can_acess_battle_list(self):
+        response = self.auth_client.get(reverse('battle_list'))
+        self.assertEqual(response.status_code, 200)
+
+    def test_view_returns_empty_list(self):
+        battle = Battle.objects.get(creator=self.user)
+        battle.delete()
+        response = self.auth_client.get(reverse('battle_list'))
+        response_qs = response.context_data.get('battle_list')
+        self.assertCountEqual(response_qs, [])
+
+    def test_view_returns_one_battle_in_list(self):
+        response = self.auth_client.get(reverse('battle_list'))
+        response_qs = response.context_data.get('battle_list')
+        self.assertCountEqual(response_qs, [self.battle_from_user][0])
+
+    def test_view_returns_few_battle_ids(self):
+        baker.make('battle.Battle', creator=self.user, _quantity=4)
+        response = self.auth_client.get(reverse('battle_list'))
+        response_qs = response.context_data.get('battle_list')
+        self.assertCountEqual(self.battle_from_user, response_qs)
+
+    def test_view_returns_a_lot_battle_ids(self):
+        baker.make('battle.Battle', creator=self.user, _quantity=100)
+        response = self.auth_client.get(reverse('battle_list'))
+        response_qs = response.context_data.get('battle_list')
+        self.assertCountEqual(self.battle_from_user, response_qs)
+
+    def test_view_an_error_when_the_user_is_not_logged(self):
+        self.auth_client.logout()
+        response = self.client.get(reverse('battle_list'))
+        self.assertRedirects(response, '/accounts/login/?next=/battle/list/')
+
+    def test_view_diff_to_verify_it_returns_exactly_len_of_battle_list(self):
+        baker.make('battle.Battle', creator=self.user, _quantity=10)
+        response_initial = self.auth_client.get(reverse('battle_list'))
+        response_qs_initial = response_initial.context_data.get('battle_list')
+
+        self.assertCountEqual(self.battle_from_user, response_qs_initial)
+
+        baker.make('battle.Battle', creator=self.user, _quantity=90)
+        battles_updated = Battle.objects.filter(creator=self.user)
+        response_updated = self.auth_client.get(reverse('battle_list'))
+        response_qs_updated = response_updated.context_data.get('battle_list')
+
+        self.assertCountEqual(battles_updated, response_qs_updated)
+
+
+class BattleCreateViewTest(TestCaseUtils):
+    def setUp(self):
+        super().setUp()
+        self.opponent = baker.make('users.User')
+
+    def test_create_battle_successfully(self):
+        battle_data = {
+            "creator": self.user.id,
+            "opponent": self.opponent.email,
+        }
+        self.auth_client.post(reverse('battle'), battle_data)
+        battle = Battle.objects.filter(creator=self.user, opponent=self.opponent)
+        self.assertTrue(battle)
```

Comment:
Please check if the creator and opponent are the same as the ones you set, and that the winner is none

Reply:
Thank you! But I have one question. When I filter by the creator and opponent I set, it doesn't do this? (talking about verifying the creator and opponent)
About the winner >> I'm going to do this :) thanks!!

Reply:
yes, ok I will let this go